contracts/staking/contracts/src/fees/MixinExchangeFees.sol

```diff
@@ -88,206 +83,116 @@ contract MixinExchangeFees is
+        if (pool.feesCollected == 0) {
+            // Compute member and total weighted stake.
+            (pool.membersStake, pool.weightedStake) = _computeMembersAndWeightedStake(poolId, poolStake);
+
+            // Increase the total weighted stake.
+            totalWeightedStakeThisEpoch = totalWeightedStakeThisEpoch.safeAdd(pool.weightedStake);
 
-        _goToNextEpoch();
+            // Increase the number of active pools.
+            numActivePoolsThisEpoch += 1;
+
+            // Emit an event so keepers know what pools to pass into
+            // `finalize()`.
+            emit StakingPoolActivated(currentEpoch_, poolId);
+        }
+
+        // Credit the fees to the pool.
+        pool.feesCollected = pool.feesCollected.safeAdd(protocolFeePaid);
+
+        // Increase the total fees collected this epoch.
+        totalFeesCollectedThisEpoch = totalFeesCollectedThisEpoch.safeAdd(protocolFeePaid);
+
+        // Store the pool.
+        activePoolsThisEpoch[poolId] = pool;
     }
 
-    /// @dev Returns the total amount of fees collected thus far, in the current epoch.
-    /// @return Amount of fees.
-    function getTotalProtocolFeesThisEpoch()
+    /// @dev Get information on an active staking pool in this epoch.
+    /// @param poolId Pool Id to query.
+    /// @return pool ActivePool struct.
+    function getActiveStakingPoolThisEpoch(bytes32 poolId)
         external
         view
-        returns (uint256)
+        returns (IStructs.ActivePool memory pool)
     {
-        uint256 wethBalance = IEtherToken(WETH_ADDRESS).balanceOf(address(this));
-        return address(this).balance.safeAdd(wethBalance);
+        pool = _getActivePoolFromEpoch(currentEpoch, poolId);
+        return pool;
     }
 
-    /// @dev Withdraws the entire WETH balance of the contract.
-    function _unwrapWETH()
-        internal
+    /// @dev Computes the members and weighted stake for a pool at the current
+    ///      epoch.
+    /// @param poolId ID of the pool.
+    /// @param totalStake Total (unweighted) stake in the pool.
+    /// @return membersStake Non-operator stake in the pool.
+    /// @return weightedStake Weighted stake of the pool.
+    function _computeMembersAndWeightedStake(
+        bytes32 poolId,
+        uint256 totalStake
+    )
+        private
+        view
+        returns (uint256 membersStake, uint256 weightedStake)
     {
-        uint256 wethBalance = IEtherToken(WETH_ADDRESS).balanceOf(address(this));
-
-        // Don't withdraw WETH if the WETH balance is zero as a gas optimization.
-        if (wethBalance != 0) {
-            IEtherToken(WETH_ADDRESS).withdraw(wethBalance);
-        }
+        uint256 operatorStake = getStakeDelegatedToPoolByOwner(
+            _poolById[poolId].operator,
+            poolId
+        ).currentEpochBalance;
+        membersStake = totalStake.safeSub(operatorStake);
+        weightedStake = operatorStake.safeAdd(
+            LibMath.getPartialAmountFloor(
```

Comment:
The `weightedStake` should also include the operator's stake. See https://github.com/0xProject/0x-monorepo/blob/3.0/contracts/staking/contracts/src/fees/MixinExchangeFees.sol#L214